generic/release-checklist.md

```diff
@@ -0,0 +1,106 @@
+---
+# Template generated by https://github.com/coreos/repo-templates; do not edit downstream
+---
+
+# Release process
+
+The release process follows the usual PR-and-review flow, allowing an external reviewer to have a final check before publishing.
+
+## Requirements
+
+This guide requires:
+
+ * A web browser (and network connectivity)
+ * `git`
+ * [GPG setup][GPG setup] and personal key for signing
+ * Write access to this GitHub project
+
+## Release checklist
+
+These steps show how to release version `x.y.z` on the `origin` remote (this can be checked via `git remote -av`).
+Push access to the upstream repository is required in order to publish the new tag and the PR branch.
+
+:warning:: if `origin` is not the name of the locally configured remote that points to the upstream git repository (i.e. `git@github.com:coreos/{{ git_repo }}.git`), be sure to assign the correct remote name to the `UPSTREAM_REMOTE` variable.
+
+- make sure the project is clean and prepare the environment:
+  - [ ] `git clean -fd`
+  - [ ] `RELEASE_VER=x.y.z`
+  - [ ] `PREVIOUS_RELEASE_VER=x.y.z-1`
+  - [ ] `UPSTREAM_REMOTE=origin`
+
+- write release notes on a dedicated branch and tag it (the commit and tag will be signed with the GPG signing key you configured):
+  - [ ] `git checkout -b release-${RELEASE_VER}`
+  - [ ] write release notes in `docs/release-notes.md`. You can follow the following template:
+    ```
+    cat <<EOF > release-notes.txt
+    This is release ${NEWTAG} of console-login-helper-messages.
+
+    Thanks to the following contributors for patches during this release:
+
+    $(git shortlog --no-merges --pretty=format:"%h %s" -e ${PREVIOUS_RELEASE_VER}..HEAD)
+    EOF
+    ```
+  - [ ] commit your changes and create the tag:
+    ```
+    git tag --file release-notes.txt -s "v${RELEASE_VER}"
+    ```
+  - [ ] verify the tag and its signature:
+    ```
+    git show "v${RELEASE_VER}"
+    git tag --verify "v${RELEASE_VER}"
+    ```
+
+- open and merge a PR for this release:
+  - [ ] `git push --tags ${UPSTREAM_REMOTE} release-${RELEASE_VER}`
+  - [ ] open a web browser and create a PR for the branch above
```

Comment:
What are we PRing here?  We've created a branch but not committed anything to it, and we've already pushed the tag so we couldn't change any commits anyway.